Assignment_03.rb

```diff
@@ -8,29 +9,23 @@
 You can use group_by, each, sum etc methods as per your need
 =end
 
+# This fucntion takes an array of hash<:id,:marks> as argument and returns the a hash containing the id mapped to another hash containing the total marks of that id.
 def group_total_marks(marks)
   unless marks.kind_of?(Array)
     return "Enter an array input"
   end
 
-  res = Hash.new
+  result = Hash.new
   temp = marks.group_by { |x| x[:id] }
-  temp.each do |k,v_arr|
-    total = 0
 
-    v_arr.each do |x|
-      total += x[:marks]
-    end
-    
-    res[k] = { total:  total }
+  temp.each do |k,v_arr|

```

Comment:
give a line space after the comma
should be `|key, val|`

and instead of `temp` and `v_arr` please give meaning full name so that it is understandable